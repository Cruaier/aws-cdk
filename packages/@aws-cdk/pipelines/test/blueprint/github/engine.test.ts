import { mkdtempSync, readFileSync } from 'fs';
import { tmpdir } from 'os';
import { join } from 'path';
import { App, Stack, Stage } from '@aws-cdk/core';
import { Pipeline, SynthStep } from '../../../lib';
import { GitHubEngine } from './engine';


test('pipeline with only a synth step', () => {
  const app = new App();

  const github = new GitHubEngine({
    workflowPath: `${mkoutdir()}/deploy.yml`,
  });

  new Pipeline(app, 'Pipeline', {
    engine: github,
    synthStep: new SynthStep('Build', {
      installCommands: ['yarn'],
      commands: ['yarn build'],
    }),
  });

  app.synth();

  expect(readFileSync(github.workflowPath, 'utf-8')).toMatchSnapshot();
});

test('single wave/stage/stack', () => {
  const app = new App();

  const github = new GitHubEngine({
    workflowPath: `${mkoutdir()}/deploy.yml`,
  });

  const stage = new Stage(app, 'MyStack');
  new Stack(stage, 'MyStack');

  const pipeline = new Pipeline(app, 'MyPipeline', {
    engine: github,
    synthStep: new SynthStep('Build', {
      installCommands: ['yarn'],
      commands: ['yarn build'],
    }),
  });

  pipeline.addStage(stage);

  app.synth();

  expect(readFileSync(github.workflowPath, 'utf-8')).toMatchSnapshot();
});

function mkoutdir() {
  return mkdtempSync(join(tmpdir(), 'cdk-pipelines-github-'));
}