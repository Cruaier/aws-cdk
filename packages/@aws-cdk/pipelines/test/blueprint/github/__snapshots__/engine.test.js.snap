// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only build 1`] = `
"name: deploy
on:
  push:
    branches:
      - main
  workflow_dispatch: {}
jobs:
  - name: Synthesize
    permissions:
      contents: read
    runsOn: ubuntu-20.0
    env: {}
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Install
        run: yarn
      - name: Build
        run: yarn build
      - name: Upload cdk.out
        uses: actions/upload-artifact@v2.1.1
        with:
          name: cdk.out
          path: cdk.out
"
`;

exports[`pipeline with only a synth step 1`] = `
"name: deploy
on:
  push:
    branches:
      - main
  workflow_dispatch: {}
jobs:
  build:
    name: Synthesize
    permissions:
      contents: read
    runsOn: ubuntu-20.0
    env: {}
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Install
        run: yarn
      - name: Build
        run: yarn build
      - name: Upload cdk.out
        uses: actions/upload-artifact@v2.1.1
        with:
          name: cdk.out
          path: cdk.out
"
`;

exports[`single wave/stage/stack 1`] = `
"name: deploy
on:
  push:
    branches:
      - main
  workflow_dispatch: {}
jobs:
  build:
    name: Synthesize
    permissions:
      contents: read
    runsOn: ubuntu-20.0
    env: {}
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Install
        run: yarn
      - name: Build
        run: yarn build
      - name: Upload cdk.out
        uses: actions/upload-artifact@v2.1.1
        with:
          name: cdk.out
          path: cdk.out
"
`;
